### DISCUSSION TITLE
Selenoid, Seleniumbase and UC Mode

### DISCUSSION THREAD

**Discussion by G-markier** [seleniumbase/SeleniumBase#2285] [Q&A] [answered]
1. Does Seleniumbase works with Selenoid for undetectable browsing? 
2. If yes, does it work in UC Mode where it can make modifications to the driver (transforming chromedriver into uc_driver)?
3. If no, do you know of a viable replacement for Selenoid that allows for real time video recording using VNC or similar?

**Comment by mdmintz** [comment, marked as answer]
1. SeleniumBase in basic mode supports Selenoid. (https://github.com/seleniumbase/SeleniumBase/discussions/1087#discussioncomment-1714005)
(But not in UC Mode, which doesn't work with Selenium Grids.)
2. -
3. Not sure. Video recording adds unnecessary overhead.

**Reply by G-markier** [reply]
Thanks.
Regarding #3, we use the built-in video recording for Selenoid to review our spiders to make sure they looking human like, the videos provide that record that also allows us to look back and try to diagnose when something goes wrong that does not give any errors.

Question: Is it the implementation of UC Mode that doesn't work with Selenium Grids, or UC itself doesn't work with Selenium grid? 

I am not a developer, so I am asking just to see if hiring a developer to integrate Selenoid with UC would be a viable option.

**Reply by mdmintz** [reply]
The original undetected-chromedriver doesn't support Selenium Grid:
https://github.com/search?q=repo%3Aultrafunkamsterdam%2Fundetected-chromedriver+grid&type=issues

Selenium Grid is a problem for UC Mode because while the Grid controls chromedriver, the Grid does not have the root control that is needed. UC Mode requires raw control for modifying a local chromedriver (renamed to uc_driver), as well as for disconnecting the driver from Chrome, and reconnecting to it a short time later.

**Reply by thule2k** [reply]
What about Selenoid? It contains the driver inside the container along with the browser, or am I wrong?

**Reply by thule2k** [reply]
Or maybe it's possible to work with pure cdp: https://aerokube.com/selenoid/latest/#_accessing_browser_developer_tools
